### PR TITLE
fix: preserve unresolved shell env vars during curl import instead of…

### LINKED ISSUE #495: [Bug] URL Block Does Not Preserve Shell Environment Variables ($VARIABLE)
Description

When a cURL command contains shell environment variables (e.g. `$AMBER_ADMIN_URL`, `$AMBER_ADMIN_USER`, `$AMBER_ADMIN_PASS`), the URL block does not preserve them correctly. The block gets distorted, making the command difficult to read and copy.

## Steps to Reproduce

1. Create or paste the following cURL command into the editor:

```bash
curl -u "$AMBER_ADMIN_USER:$AMBER_ADMIN_PASS" \
  -X POST "$AMBER_URL/admin/consumers/<ORG_ID>/atoms" \
  -H 'Content-Type: application/json' \
  -d '{"atoms": 123456778, "topup_id": "free"}'
```
Observe how the URL block renders the command

### PR DESCRIPTION
**Pull request by rajarshi1**
Fixes #495

### Problem
When importing a curl command that references shell environment variables (e.g. `$AMBER_ADMIN_URL`), the URL/header/body gets visibly distorted. Example from the issue:

    curl -u "$AMBER_ADMIN_USER:$AMBER_ADMIN_PASS" \
      -X POST "$AMBER_URL/admin/consumers/<ORG_ID>/atoms" \
      -H 'Content-Type: application/json' \
      -d '{"atoms": 123456778, "topup_id": "free"}'

### Root cause
`curl.ts` always calls the shell-quote parser with no `env` map:

    const parseEntries = parse(rawData.replace(/\\n/g, " "), undefined, undefined);

Inside `getVar()`, any `$VAR` with no matching entry in the (empty) env map silently became an **empty string** rather than staying as literal text. For the reported case, `$AMBER_URL/admin/consumers/<ORG_ID>/atoms` collapsed to just `/admin/consumers/<ORG_ID>/atoms` — the whole base URL vanished.

Confirmed via search that `curl.ts` is the only caller of this parser, and it always passes `env=undefined` — so there's no other code path relying on the old "blank out unknown vars" behavior.

### Fix
`getVar()` now preserves the original `$VAR` / `${VAR}` syntax when a variable is unresolved, instead of stripping it. Also threaded a `braced` flag through `parseEnvVar()` so `${VAR}` and `$VAR` each reconstruct in their original form (not just collapsed to one style), avoiding any ambiguity like `${FOO}BAR` incorrectly becoming `$FOOBAR`.

Real values still substitute correctly when an env map *is* supplied — this only changes what happens for genuinely unresolved variables.

### Testing
New test file: `apps/ui/src/core/lib/parser/importers/__test__/shell-quote-override.test.ts`

- 7/7 new tests passing, including a direct reproduction of the issue's exact URL
- Full `voiden-ui` suite: 184/185 passing (the 1 failure is a pre-existing, unrelated `pipeline.test.ts` assertion — confirmed identical on clean `main` with this diff removed)

### Files changed
- `apps/ui/src/core/lib/parser/importers/shell-quote-override.ts`
- `apps/ui/src/core/lib/parser/importers/__test__/shell-quote-override.test.ts` (new)

### CODE REVIEW

File: apps/ui/src/core/lib/parser/importers/__test__/shell-quote-override.test.ts
Comment (A-S-Manoj) [inline apps/ui/src/core/lib/parser/importers/__test__/shell-quote-override.test.ts:1]: These tests only cover parse() in isolation — they don't exercise convert() in curl.ts, which is the actual code path from the linked issue. Could you add an end-to-end test that calls convert(rawData) with the issue's exact command and asserts on the final url/authentication fields? That would have caught the stripping-loop interaction above.

File: apps/ui/src/core/lib/parser/importers/shell-quote-override.ts [hunk 1]
```diff
@@ -87,10 +87,19 @@ function matchAll(s, r) {
...
-    r = "";
+    // @custom start
+    // Preserve the original $VAR / ${VAR} syntax instead of silently
+    // stripping it to an empty string. Voiden's curl importer always calls
+    // parse() with no env map (env=undefined -> {} here), since it's
+    // importing a pasted command, not executing it in a real shell. Blanking
+    // out unresolved shell variables (e.g. $AMBER_ADMIN_URL) distorts the
+    // resulting URL/header/body instead of preserving the reference for the
+    // user to see and map to a Voiden environment variable afterwards.
+    r = braced ? "${" + key + "}" : "$" + key;
```
Comment (A-S-Manoj) [inline apps/ui/src/core/lib/parser/importers/shell-quote-override.ts:101]: Good fix here — but `curl.ts` has an untouched post-processing step that strips a leading `$` and will undo this for real cases. Details in my summary comment.

**Review by A-S-Manoj — CHANGES_REQUESTED** [review]
`getVar()` correctly preserves unresolved `$VAR`/`${VAR}` at the parser level — nice fix, tests look good.

But it doesn't fully solve the issue. `convert()` in `curl.ts` (untouched here) has an old loop that strips a leading `$` from string tokens:

https://github.com/VoidenHQ/voiden/blob/438eb79/apps/ui/src/core/lib/parser/importers/curl.ts#L379-L384

That was a no-op before (unresolved vars used to become `""`), but now it fires on the preserved `$AMBER_URL/...` tokens and strips just the `$`. I ran `convert()` on the issue's repro command and got `url: "http://amber_url/admin/consumers/%3CORG_ID%3E/atoms"` and inconsistent auth (`username: "AMBER_ADMIN_USER"` vs `password: "$AMBER_ADMIN_PASS"`).

Since `curl.ts` is the only caller of this parser, this is the same path the issue reports, not an edge case. Can you remove/guard that loop, or add an end-to-end test via `convert()`? Requesting changes for now — parser fix itself is solid.

**Comment by rajarshi1** [comment]
Good catch — you were right, this wasn't fully solved. Confirmed and fixed.

**Root cause:** `curl.ts`'s `convert()` had its own untouched post-processing loop stripping a leading `$` from every string token. Before the parser fix, no token could ever legitimately start with `$` (every `$` was already consumed during parsing), so it was dead code. After the fix, it silently re-mangled the exact variables the parser fix preserves.

I actually ran `convert()` on your repro command and reproduced your exact output before touching anything:

    url: "http://amber_url/admin/consumers/%3CORG_ID%3E/atoms"
    username: "AMBER_ADMIN_USER"
    password: "$AMBER_ADMIN_PASS"

Removed the stripping loop in `curl.ts` — it now just passes the token through as parsed. Same repro command after the fix:

    url: "http://$amber_url/admin/consumers/%3CORG_ID%3E/atoms"
    username: "$AMBER_ADMIN_USER"
    password: "$AMBER_ADMIN_PASS"

Both auth fields now consistent. (Host shows lowercase `$amber_url` — that's the `URL` API's own hostname normalization, case-insensitive by spec, unrelated to either fix.)

Added `__test__/curl.test.ts` with 5 tests calling `convert()` directly (not `parse()` in isolation), including your exact repro command asserted field-by-field, plus a check that a literal non-variable `$5` isn't mistaken for the same case.

Full suite: 189/190 passing, the 1 failure is pre-existing/unrelated (confirmed identical on `main` with this diff removed). `tsc --noEmit` also confirmed clean on both new/modified files.

**Review by A-S-Manoj — APPROVED** [review]
Fix looks good — confirmed the curl.ts change resolves the stripping issue, and tests now cover convert() end-to-end. Approving. Maintainers can take it from here. Good luck on merge.

**Comment by phurpa-tsering** [comment]
Good fix @rajarshi1 @A-S-Manoj — but heads up: apps/ui/src/core/lib/parser/importers/curl.ts (the file this patches) looks like dead code. The real "paste a cURL command" flow runs through a separate copy in the [voiden-rest-api plugin](https://github.com/VoidenHQ/plugin-voiden-rest-api), which already has this exact fix shipped in v1.4.7.

Safe to merge though — won't affect anything either way, and the tests are a nice addition regardless.

**Review by phurpa-tsering — APPROVED** [review]
